uefi/section.go

```diff
@@ -71,18 +84,32 @@ type SectionExtHeader struct {
 	ExtendedSize uint32 `json:"-"`
 }
 
+// SectionGUIDDefined contains the fields for a EFI_SECTION_GUID_DEFINED
+// section.
+type SectionGUIDDefined struct {
+	GUID       uuid.UUID
+	DataOffset uint16
+	Attributes uint16
+}
+
 // Section represents a Firmware File Section
 type Section struct {
 	Header SectionExtHeader
 	Type   string
 	buf    []byte
 
-	//Metadata for extraction and recovery
+	// Metadata for extraction and recovery
 	ExtractPath string
 	fileOrder   int
 
 	// Type specific fields
+	TypeSpecific interface{} `json:",omitempty"`
+
+	// For EFI_SECTION_USER_INTERFACE
 	Name string `json:",omitempty"`
+
+	// Encapsulated section (only for EFI_SECTION_COMPRESSION, EFI_SECTION_GUID_DEFINED and EFI_SECTION_DISPOSABLE)
+	Encapsulated *Section `json:",omitempty"`
```

Comment:
Are you sure it's always going to be a Section? I think it can be other things. you should probably use []Firmware

You also don't want to assume there's only one thing.